src/content/docs/en/guides/cms/datocms.mdx

```diff
@@ -0,0 +1,195 @@
+	method: 'POST',
+	headers: {
+		'Content-Type': 'application/json',
+		Accept: 'application/json',
+		Authorization: `Bearer ${import.meta.env.DATOCMS_API_KEY}`,
+	},
+	body: JSON.stringify({
+		query: `query Homepage {
+          home {
+            title
+          }
+        }
+      `,
+	}),
+});
+
+const json = await response.json();
+const data = json.data.home;
+---
+
+<h1>{data.title}</h1>
+
+```
+
+This GraphQL query will fetch the "title" field in the "home" page from your DatoCMS Project. When you refresh your browser, you should see the title on your page.
+
+## Dynamically fetch blocks
+
+If you want to dynamically build your page in DatoCMS and support the rearrangement of blocks, you can create a "Modular Content" block in the homepage. [Read more about modular content here](https://www.datocms.com/docs/content-modelling/modular-content).
+
+In order to get the new blocks, you'll need to tweak the GraphQL query a bit. To check which block the code need to display, the query also includes the unique `_modelApiKey` of each item. Make sure to create an Astro component for each block you made in DatoCMS to correctly display the data.
+
+```astro title="src/pages/index.astro"
+---
+import Image from '../components/Image.astro';
+import Text from '../components/Text.astro';
+
+const response = await fetch('https://graphql.datocms.com/', {
+	method: 'POST',
+	headers: {
+		'Content-Type': 'application/json',
+		Accept: 'application/json',
+		Authorization: `Bearer ${import.meta.env.DATOCMS_API_KEY}`,
+	},
+	body: JSON.stringify({
+		query: `query Homepage {
+          home {
+            title
+            content {
+              ... on ImageRecord {
+                _modelApiKey
+               image{
+                url
+               }
+              }
+              ... on TextRecord {
+                _modelApiKey
+                text(markdown: true)
+              }
+            }
+          }
+        }
+      `,
+	}),
+});
+
+const json = await response.json();
+const data = json.data.home;
+---
+
+<h1>{data.title}</h1>
+{
+  data.content.map((item: any) => {
+    switch (item._modelApiKey) {
+      case 'image':
+        return <Image item={item} />;
+      case 'text':
+        return <Text item={item} />;
+      default:
+        return null;
+    }
+  })
+}
+
+```
+
+The example below will show you how to create a component for a "Multiple-paragraph text" block from DatoCMS. The above GraphQL query requests the markdown data, which returns a plain HTML string, which you need to dangerously set.
+
+```astro title="src/components/Text.astro"
+---
+export interface TextProps {
+	text: string
+}
+
+export interface Props {
+	item: TextProps
+}
+
+const { item } = Astro.props;
+---
+
+<div set:html={item.text} />
+
+```
+
+## Publishing your site
+
+To deploy your website, visit our [deployment guides](https://docs.astro.build/en/guides/deploy/) and follow the instructions for your preferred hosting provider.
```

Comment:
```suggestion
To deploy your website, visit our [deployment guides](/en/guides/deploy/) and follow the instructions for your preferred hosting provider.
```

Reply:
I'm gonna go ahead and merge this one, since this will make the CI all green for ya! ✅